src/Tools/dotnet-monitor/Egress/EgressProviderInternal.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress
+{
+    /// <summary>
+    /// A wrapper implementation that takes an egress provider name, gets the egress provider options
+    /// based on that name, and invokes the corresponding <see cref="IEgressProvider{TOptions}"/> implementation
+    /// to perform the egress action.
+    /// </summary>
+    internal class EgressProviderInternal<TOptions> :
+        IEgressProviderInternal<TOptions>
+        where TOptions : class
+    {
+        private readonly ILogger<EgressProviderInternal<TOptions>> _logger;
+        private readonly IEgressProvider<TOptions> _provider;
+        private readonly IDynamicOptionsSource<TOptions> _source;
+        private readonly IEnumerable<IPostConfigureOptions<TOptions>> _postConfigures;
+        private readonly IValidateOptions<TOptions> _validation;
+
+        public EgressProviderInternal(
+            ILogger<EgressProviderInternal<TOptions>> logger,
+            IEgressProvider<TOptions> provider,
+            IDynamicOptionsSource<TOptions> source,
+            IEnumerable<IPostConfigureOptions<TOptions>> postConfigures,
+            IValidateOptions<TOptions> validatation)
```

Comment:
I realize it doesn't _really_ matter, but just since I noticed it there's a typo -> validatation instead of validation. It just stood out when reading the line _validation = validation;

Reply:
I think spelling matters, so good catch! I always welcome spelling and grammar corrections.